great_expectations/data_context/data_context/abstract_data_context.py

```diff
@@ -338,13 +338,18 @@ def __init__(self, runtime_environment: Optional[dict] = None) -> None:
         self._attach_fluent_config_datasources_and_build_data_connectors(
             self.fluent_config
         )
-        oss_id = self._get_oss_id()
         self._anonymizer = Anonymizer(salt=self._data_context_id)
+
+        # TODO: Need to determine V1 mechanism for usage stats opt-out
+        if self._is_usage_stats_enabled():
```

Comment:
You shouldn't do this check here as we no longer want to be gated by this environment variable. The call to `_init_analytics` initializes the posthog client. If `GX_ANALYTICS_ENABLED` is set to False this will disable the client. I don't know the behavior of `submit_event` if posthog.enabled == `false` but presumable, it is a no-op. @wookasz  can you verify? 
I would experiment to understand the behavior here.

If not, you should change the init_analytics call to return a value that let's us know if analytics is enable or not.